tests/cloudpickle_test.py

```diff
@@ -2351,6 +2363,225 @@ def __type__(self):
+                import mock_local_folder.subfolder.submod
+                register_pickle_by_value(mock_local_folder.subfolder.submod)
+                assert w.run(lambda: _func()) == _func()
+                assert w.run(lambda: _t.__name__) == _t.__name__
+                assert w.run(lambda: _class().method()) == _class().method()
+
+                # Registering a module from a subfolder for pickling by value
+                # should not make constructs from modules from the parent
+                # folder pickleable
+                with pytest.raises(ImportError):
+                    w.run(lambda: local_function)
+                with pytest.raises(ImportError):
+                    w.run(lambda: __import__("mock_local_folder.mod"))
+
+                unregister_pickle_by_value(
+                    mock_local_folder.subfolder.submod
+                )
+                with pytest.raises(ImportError):
+                    w.run(lambda: local_submod_function)
+
+                # Test the subfolder namespace module case
+                import mock_local_folder.subfolder
+                register_pickle_by_value(mock_local_folder.subfolder)
+                assert w.run(lambda: _func()) == _func()
+                assert w.run(lambda: _t.__name__) == _t.__name__
+                assert w.run(lambda: _class().method()) == _class().method()
+
+                unregister_pickle_by_value(mock_local_folder.subfolder)
+        finally:
+            _fname = "mock_local_folder"
+            sys.path = _prev_sys_path
+            for m in [_fname, f"{_fname}.mod", f"{_fname}.subfolder",
+                      f"{_fname}.subfolder.submod"]:
+                mod = sys.modules.pop(m, None)
+                if mod and mod.__name__ in list_registry_pickle_by_value():
+                    unregister_pickle_by_value(mod)
+
+    def test_pickle_constructs_from_installed_packages_registered_for_pickling_by_value(  # noqa
+        self
+    ):
+        for package_or_module in ["package", "module"]:
+            if package_or_module == "package":
+                import _cloudpickle_testpkg as m
+                f = m.package_function_with_global
+                _original_global = m.global_variable
+            elif package_or_module == "module":
+                import _cloudpickle_testpkg.mod as m
+                f = m.module_function_with_global
+                _original_global = m.global_variable
+            try:
+                with subprocess_worker(protocol=self.protocol) as w:
+                    assert w.run(lambda: f()) == _original_global
+
+                    # Test that f is pickled by value by modifying a global
+                    # variable that f uses, and making sure that this
+                    # modification shows up when calling the function remotely
+                    register_pickle_by_value(m)
+                    assert w.run(lambda: f()) == _original_global
+                    m.global_variable = "modified global"
```

Comment:
Just to make sure the test does not trivially pass:

```suggestion
                    m.global_variable = "modified global"
                    assert m.global_variable != _original_global
```